src/t_stream.c

```diff
@@ -4890,6 +5037,122 @@ NULL
     }
 }
 
+/* XIDMP CFGGET <key> [DURATION] [MAXSIZE]
+ * XIDMP CFGSET <key> [DURATION <duration>] [MAXSIZE <maxsize>] */
+void xidmpCommand(client *c) {
+    char *opt = c->argv[1]->ptr;
+    robj *key = c->argv[2];
+    
+    /* Lookup the stream key */
+    kvobj *kv = lookupKeyReadOrReply(c, key, shared.nokeyerr);
+    if (kv == NULL || checkType(c, kv, OBJ_STREAM)) return;
+    stream *s = kv->ptr;
+    if (!strcasecmp(opt, "CFGGET")) {
+        /* XIDMP CFGGET <key> [DURATION] [MAXSIZE] */
+        int get_duration = 0;
+        int get_maxsize = 0;
+        
+        /* If no parameters specified, return both */
+        if (c->argc == 3) {
+            get_duration = 1;
+            get_maxsize = 1;
+        } else {
+            /* Parse which parameters to get */
+            for (int i = 3; i < c->argc; i++) {
+                char *param = c->argv[i]->ptr;
+                if (!strcasecmp(param, "DURATION")) {
+                    get_duration = 1;
+                } else if (!strcasecmp(param, "MAXSIZE")) {
+                    get_maxsize = 1;
+                } else {
+                    addReplyErrorFormat(c, "Unknown parameter '%s'", param);
+                    return;
+                }
+            }
+        }
+        
+        /* Build reply as a map */
+        int num_fields = get_duration + get_maxsize;
+        addReplyMapLen(c, num_fields);
+        if (get_duration) {
+            addReplyBulkCString(c, "duration");
+            addReplyLongLong(c, s->idmp_duration);
+        }
+        if (get_maxsize) {
+            addReplyBulkCString(c, "maxsize");
+            addReplyLongLong(c, s->idmp_max_entries);
+        }
+    } else if (!strcasecmp(opt, "CFGSET")) {
+        /* XIDMP CFGSET <key> [DURATION <duration>] [MAXSIZE <maxsize>] */
+        long long duration = -1;
+        long long maxsize = -1;
+        int duration_set = 0;
+        int maxsize_set = 0;
+        
+        /* Parse parameters */
+        for (int i = 3; i < c->argc; i++) {
+            int moreargs = c->argc - i - 1;
+            char *param = c->argv[i]->ptr;
+            if (!strcasecmp(param, "DURATION") && moreargs) {
+                if (duration_set) {
```

Comment:
what about, in this case, duration_set is not needed.
```suggestion
                if (duration != -1) {
```

Reply:
Done.